compiler/luci-interpreter/src/kernels/Mean.cpp

```diff
@@ -0,0 +1,245 @@
+{
+}
+
+void Mean::configure()
+{
+  assert(_input->element_type() == _output->element_type());
+  assert(_reduction_indices->element_type() == DataType::S32);
+  const Shape input_shape = _input->shape();
+  int input_num_dims = input_shape.num_dims();
+
+  const auto *axis = getTensorData<int32_t>(_reduction_indices);
+  int num_axis = _reduction_indices->shape().num_elements();
+  assert(num_axis <= 4);
+
+  if (input_num_dims == 0)
+  {
+    _output->resize(Shape(0));
+    return;
+  }
+
+  if (_params.keep_dims)
+  {
+    Shape output_shape(input_num_dims);
+    for (int idx = 0; idx < input_num_dims; ++idx)
+    {
+      bool is_axis = false;
+      for (int axis_idx = 0; axis_idx < num_axis; ++axis_idx)
+      {
+        if (axis[axis_idx] == idx || axis[axis_idx] + input_num_dims == idx)
+        {
+          is_axis = true;
+          break;
+        }
+      }
+      if (is_axis)
+      {
+        output_shape.dim(idx) = 1;
+      }
+      else
+      {
+        output_shape.dim(idx) = input_shape.dim(idx);
+      }
+    }
+    _output->resize(output_shape);
+    return;
+  }
+  else
+  {
+    // Calculates size of reducing axis.
+    int num_reduce_axis = num_axis;
+    for (int i = 0; i < num_axis; ++i)
+    {
+      int current = axis[i];
+      if (current < 0)
+      {
+        current += input_num_dims;
+      }
+      assert(current >= 0 && current < input_num_dims);
+      for (int j = 0; j < i; j++)
+      {
+        int previous = axis[j];
+        if (previous < 0)
+        {
+          previous += input_num_dims;
+        }
+        if (current == previous)
+        {
+          --num_reduce_axis;
+          break;
+        }
+      }
+    }
+    // Determines output dimensions.
+    Shape output_shape(input_num_dims - num_reduce_axis);
+    int num_skip_axis = 0;
+    for (int idx = 0; idx < input_num_dims; ++idx)
+    {
+      bool is_axis = false;
+      for (int axis_idx = 0; axis_idx < num_axis; ++axis_idx)
+      {
+        if (axis[axis_idx] == idx || axis[axis_idx] + input_num_dims == idx)
+        {
+          ++num_skip_axis;
+          is_axis = true;
+          break;
+        }
+      }
+      if (!is_axis)
+      {
+        output_shape.dim(idx - num_skip_axis) = input_shape.dim(idx);
+      }
+    }
+    _output->resize(output_shape);
+  }
+}
+
+void Mean::execute() const
+{
+  switch (_input->element_type())
+  {
+    case DataType::FLOAT32:
+      evalFloat();
+      break;
+    case DataType::U8:
+      evalQuantized();
+      break;
+    default:
+      throw std::runtime_error("Unsupported type.");
+  }
+}
+
+void Mean::evalFloat() const
+{
+  const Shape input_shape = _input->shape();
+  int input_num_dims = input_shape.num_dims();
+  const auto *axis = getTensorData<int32_t>(_reduction_indices);
+  int num_axis = _reduction_indices->shape().num_elements();
+
+  // TODO
+  // Declare these temporary tensors somewhere else?
+  Tensor *temp_index = new Tensor(DataType::S32, Shape(input_num_dims), {}, "");
```

Comment:
I'm not sure where to declare and initialize these temporary tensors. Help needed.